### PR TITLE
Make the native viewer an optional dependency of rerun, re_sdk, rerun_py

### PR DESCRIPTION
**Pull request by emilk**
Closes https://github.com/rerun-io/rerun/issues/1384

This adds a new `native_viewer` feature-flag to `rerun_py`, `rerun` and `re_sdk`.

### Checklist
* [x] I have read and agree to [Contributor Guide](../CONTRIBUTING.md) and the [Code of Conduct](../CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)
* [ ] I've added a line to `CHANGELOG.md` (if this is a big enough change to warrant it)

### CODE REVIEW

**Comment by emilk** [comment]
Perhaps we should also rename the `web` feature to `web_viewer` for symmetry

**Comment by nikolausWest** [comment]
This feels like it also deserves a "How-to Guide"

File: crates/rerun/Cargo.toml [hunk 1]
```diff
@@ -31,19 +31,23 @@ glam = ["re_sdk?/glam"]
...
-## Embed the Rerun SDK and re-export all of its public symbols.
-sdk = ["dep:re_sdk", "re_sdk?/re_viewer"]
+## Support a native viewer
+native_viewer = ["re_sdk?/native_viewer"]
 
 ## Support for running a HTTP server that listens to incoming log messages from a Rerun SDK.
 server = ["re_sdk_comms/server"]
 
+## Embed the Rerun SDK and re-export all of its public symbols.
+sdk = ["dep:re_sdk"]
```
Comment (Wumpf) [inline crates/rerun/Cargo.toml:41]: [not actionable in _this_ PR] calling this sdk here gives us a bit of friction: In the Docs we talk about the "Rerun SDK" as the union of API and viewer. It follows we should call it re_api 🤔

File: crates/rerun/src/run.rs [hunk 2]
```diff
@@ -218,14 +230,27 @@ async fn run_impl(call_source: CallSource, args: Args) -> anyhow::Result<()> {
...
+                    call_source.app_env(),
+                    startup_options,
+                    profiler,
+                    url_or_path.clone(),
+                );
+
+                #[cfg(not(feature = "native_viewer"))]
+                {
+                    _ = call_source;
+                    anyhow::bail!("rerun compiled without the 'native_viewer' feature");
```
Comment (Wumpf) [inline crates/rerun/src/run.rs:251]:
```suggestion
                    anyhow::bail!("Can't start viewer - rerun was not with the 'native_viewer' feature");
```

File: rerun_py/src/python_bridge.rs [hunk 3]
```diff
@@ -118,7 +118,7 @@ fn rerun_bindings(py: Python<'_>, m: &PyModule) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(is_enabled, m)?)?;
     m.add_function(wrap_pyfunction!(set_enabled, m)?)?;
 
-    #[cfg(feature = "re_viewer")]
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply rerun_py/src/python_bridge.rs:121]: this is `#ifdef THING_WITH_TYPO` from C/C++ all over again :(

**Review by Wumpf — APPROVED** [review]
I'd appreciate `web_viewer` :)

**Pull request merged**